server/src/main/java/io/druid/curator/discovery/DiscoveryModule.java

```diff
@@ -476,4 +507,34 @@ public void close() throws IOException
       // nothing
     }
   }
+
+  private static class DruidLeaderSelectorProvider implements Provider<DruidLeaderSelector>
```

Comment:
usually to ease testing there is a constructor annotated with `@Inject` that populates `private final` fields